static/src/javascripts/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,263 @@
+            minAbove: 100,
+            minBelow: 100,
+        },
+        ' .element-image': {
+            minAbove: 50,
+            minBelow: 50,
+        },
+        ' .player': {
+            minAbove: 0,
+            minBelow: 0,
+        },
+        ' > h1': {
+            minAbove: 0,
+            minBelow: 0,
+        },
+        ' > h2': {
+            minAbove: 0,
+            minBelow: 0,
+        },
+        ' > *:not(p):not(h2):not(blockquote)': {
+            minAbove: 0,
+            minBelow: 0,
+        },
+        ' .ad-slot': {
+            minAbove: 100,
+            minBelow: 100,
+        },
+    },
+};
+
+// we can write a survey into a spare space using spaceFiller
+const inArticleWriter = (survey, surveyId) =>
+    spaceFiller.fillSpace(spacefinderRules, paras => {
+        const componentName = `data_tailor_survey_${surveyId}`;
+
+        mediator.emit('register:begin', componentName);
+        bonzo(survey).insertBefore(paras[0]);
+        mediator.emit('register:end', componentName);
+
+        return surveyId;
+    });
+
+// the main function to render the survey
+const renderQuickSurvey = () => {
+    const queryParams = {};
+
+    queryParams.edition = config.page.edition;
+
+    // If we want to force tailor to show a particular survey we can set an attribute in local storage to have
+    // key = 'surveyToShow', and value = the survey id. Tailor will then override other logic for display, and
+    // look for a survey with this ID to return. This is useful as we can easily see how a particular survey
+    // would be rendered, without actually putting it live. If this parameter is empty or not specified, tailor
+    // behaves as usual.
+    const surveyToShow = local.get('surveyToShow');
+
+    if (surveyToShow) {
+        queryParams.surveyToShow = surveyToShow;
+    }
+
+    // get the list of surveys that can't be shown as they have been shown recently
+    const surveysNotToShow = getSurveyIdsNotToShow();
+
+    if (surveysNotToShow) {
+        queryParams.surveysNotToShow = surveysNotToShow;
+    }
+
+    return tailor.getSuggestedSurvey(queryParams).then(suggestion => {
+        if (suggestion) {
+            storeSurveyShowedInCookie(suggestion.data);
+
+            const json = getJsonFromSurvey(suggestion.data.survey);
+            const survey = bonzo.create(template(tailorSurveyHtml, json));
+
+            return inArticleWriter(survey, suggestion.data.survey.surveyId);
+        }
+        Promise.resolve();
+    });
+};
+
+const disableRadioButtons = buttonClassName => {
+    const radioButtons = document.getElementsByClassName(buttonClassName);
+    bonzo(radioButtons).each(button => {
+        // eslint-disable-next-line no-param-reassign
+        button.disabled = true;
+    });
+};
+
+const surveyFadeOut = () => {
+    const surveyContent = document.getElementsByClassName(
+        'impressions-survey__content'
+    );
+
+    surveyContent[0].classList.add('js-impressions-survey__fadeout');
```

Comment:
[Array access is unsafe](https://flow.org/en/docs/types/arrays/#toc-array-access-is-unsafe)

> In order to make this safe, Flow would have to mark every single array access as “possibly undefined”.
> Flow does not do this because it would be extremely inconvenient to use. You would be forced to refine the type of every value you get when accessing an array.